runtime/rasdump/trigger.c

```diff
@@ -604,14 +604,13 @@ prepareForDump(struct J9JavaVM *vm, struct J9RASdumpAgent *agent, struct J9RASdu
 
 		/* Share exclusive access when it's a "slow entry" or "user" event, as there may be a deadlock */
 		UDATA shareVMAccess = exclusiveHeld &&
-				( (context->eventFlags & J9RAS_DUMP_ON_USER_SIGNAL) ||
-				(context->eventFlags & J9RAS_DUMP_ON_SLOW_EXCLUSIVE_ENTER) );
+				OMR_ARE_ANY_BITS_SET(context->eventFlags, J9RAS_DUMP_ON_USER_SIGNAL | J9RAS_DUMP_ON_SLOW_EXCLUSIVE_ENTER);
 
 		if ( shareVMAccess == 0 ) {
 
 			/* Deferred attach of SigQuit thread, needed if we're preparing to walk the heap (GC pre-req) */
-			if ( (agent->requestMask & (J9RAS_DUMP_DO_PREPARE_HEAP_FOR_WALK | J9RAS_DUMP_DO_COMPACT_HEAP | J9RAS_DUMP_DO_ATTACH_THREAD)) &&
-			(context->eventFlags & J9RAS_DUMP_ON_USER_SIGNAL ) ) {
+			if (OMR_ARE_ANY_BITS_SET(agent->requestMask, J9RAS_DUMP_DO_PREPARE_HEAP_FOR_WALK | J9RAS_DUMP_DO_COMPACT_HEAP | J9RAS_DUMP_DO_ATTACH_THREAD) &&
+					OMR_ARE_ANY_BITS_SET(context->eventFlags, J9RAS_DUMP_ON_USER_SIGNAL | J9RAS_DUMP_ON_USER2_SIGNAL)) {
```

Comment:
I noticed the thread name as well but I was hesitant to make any default changes in case a customer might be searching for this thread name (e.g. excluding it from results).